File: WoofIOS/SwiftPackages/Data/DataSource/Sources/DataSourceFakes/Moderation/ModerationDataSourceFactory.swift
import DataSource
import Swinject
import SwinjectAutoregistration

public final class ModerationDataSourceFactory {
    private let dataSource: FakeModerationDataSource

    public init(_ container: Container) {
        dataSource = container~>
    }

    @discardableResult
    public func withBlockDogError() -> Self {
        dataSource.blockDogError = .unavailable("Could not block dog")
        return self
    }
}

File: WoofAndroid/data/datasource/src/testFixtures/kotlin/com/perrystreet/woof/datasource/moderation/ModerationDataSourceFactory.kt
package com.perrystreet.woof.datasource.moderation

import org.koin.test.KoinTest
import org.koin.test.inject

class ModerationDataSourceFactory : KoinTest {
    private val dataSource: FakeModerationDataSource by inject()

    fun withBlockDogError() = apply {
        dataSource.blockDogError = IllegalStateException("Could not block dog")
    }
}
